src/system/user.cr

```diff
@@ -11,6 +11,11 @@ struct System::User
     #UID_MAX = 0xffff_u32 # POSIX Default
   {% end %}
 
+  private GETPW_R_SIZE_MAX = begin
+    size = LibC.sysconf(LibC::SC_GETPW_R_SIZE_MAX)
+    size < 0 ? 16384 : size
```

Comment:
What's the significance of `16384`? This magic number should perhaps be documented.

Reply:
@straight-shoota - Its arbitrary and was one of the samples provided by a [linux man page](http://man7.org/linux/man-pages/man3/getpwnam.3.html). One of the other [recommendations I found](https://github.com/h2o/h2o/issues/160) was to wrap the `getpwuid_r` in a loop that catches [ERANGE] and bump the amount by 1024 until it succeeds.

To get an idea of how it it could be handled I grep'ed OpenBSD's base and ports. OpenBSD sets the buffer to 1024, and lets the error system handle the rest in a number of instances. It doesn't even check `__SC_GETPW_R_SIZE_MAX` makes sense I guess. Don't trust configurable variables. Set it reasonably and error if something is wrong. Ports contained no use of `__SC_GETPW_R_SIZE_MAX`.

There are three options that I can think of should the `sysconf()` call fails:
- Set it to an arbitrary large value that is unlikely to fail i.e. 16384
- Set it to a reasonable value and raise if it fails i.e 1024
- Loop and increment the buffer until it succeeds
- Raise

I am of the opinion we should try and get `__SC_GETPW_R_SIZE_MAX` because crystal supports a number of platforms and we should first try and conform to POSIX.1. Should that fail we should resort to a buffer of 1024 bytes like the example in the [OpenGroup Spec](http://pubs.opengroup.org/onlinepubs/9699919799/functions/getpwnam.html) and as they do In OpenBSD. If that fails raise.

Thoughts? @ysbaddaden, @RX14, @asterite, @straight-shoota

Reply:
I'm fine with 16kB its only 4 pages.